packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -88,6 +89,10 @@ function writeLocalStorage(key: string, value: string) {
 export class OdspDocumentService implements IDocumentService {
     protected updateUsageOpFrequency = startingUpdateUsageOpFrequency;
 
+    readonly policies = {
```

Comment:
@vladsud Better?

Reply:
How does the developer set the policies if it's readonly?

Reply:
I feel like policies are constant thing, i.e. do not change through lifetime of the driver. I'd keep it readonly, but change it when the need arises in the future

Reply:
Oh icic... I missed the fact that prefetch happens by wrapping the driver in a PrefetchDocumentStorageService in the Container. I thought the logic was contained in the OdspDocumentStorageService itself...

New question... Do you expect the hosting application to be able to override this value or do you expect the Driver to always specify the policy? I'm not sure how the Hosting Application could set the caching policy to false if it's readonly and you don't allow value injection through the constructor.

I might just be missing the scenario

Reply:
That's a good question. I want to say that host should be able to overwrite policies, but I'd say host is more interested in end-to-end behavior of the system, not driver individually. As in changing driver policy will not do much - Container would start cacihing.
The other interesting point is that ODSP sort of caches it all the time via TreesLatest call. Host can actually overwrite it, but it does not reflect on that policy in any way.
So it feels like we need more end-to-end design for end-to-end caching policy decisions / API / etc.
I'm a little bit afraid to start big thing, given this is a patch. I'll open an issue to follow up

Reply:
https://github.com/microsoft/FluidFramework/issues/4065

Reply:
This will be a longer discussion, so resolving this comment for now.

Reply:
I just want to articulate this PR before we get it in. 

In this PR we are adding policies which allows the driver to specific things that the Container can use to infer things. The first policy we are adding is caching which will have a value of None or Prefetch.

In this PR we are updating the ODSP driver to set this value to Prefetch to keep things the same as they were before.

In the future the Fluid Server will be able to update the policy. Currently, with this PR, there will be no way for the ODSP driver to have a policy of `caching:None`. This is intentional?

Reply:
Don't want to add noise, but like Skyler want to layout what I think is the basis for this PR (CC @vladsud):

1) I think the point is to find some cheap way to remove prefetch from ODSP driver without making code changes in other drivers (maybe they're easy to make) - cheapest way is to add some boolean check that will only fail for ODSP drivers during the prefetch step
2) I am still assuming "default" behavior should be to prefetch
3) Given 1+2, the changes I should make are the ones so that only ODSP disables prefetch --  so in container.ts in getDocumentStorageService() I should only wrap storageService in PrefetchDocumentStorageService if there is no caching policy that is "None", or restated: if caching policy is "None" (as it should be in ODSP) then return the service immediately otherwise (for now) always wrap it with PrefetchDocumentService().

Reply:
That looks correct!
Sorry if it was not the same and I missed it, but latest iteration seems correct:

        // Enable prefetching for the service unless it has a caching policy set otherwise:
        if (this.service.policies?.caching === ServiceCachingPolicy.None) {
            return storageService;
        }